src/dbnode/storage/namespace.go

```diff
@@ -791,6 +793,47 @@ func (n *dbNamespace) QueryIDs(
 	return res, err
 }
 
+func (n *dbNamespace) WideQueryIDs(
+	ctx context.Context,
+	query index.Query,
+	opts index.WideQueryOptions,
+) error {
+	ctx, sp, sampled := ctx.StartSampledTraceSpan(tracepoint.NSWideQueryIDs)
+	if sampled {
+		sp.LogFields(
+			opentracinglog.String("query", query.String()),
+			opentracinglog.String("namespace", n.ID().String()),
+			opentracinglog.Int("batchSize", opts.BatchSize),
+			xopentracing.Time("start", opts.StartInclusive),
+			xopentracing.Time("end", opts.EndExclusive),
+		)
+	}
+	defer sp.Finish()
+
+	callStart := n.nowFn()
+	if n.reverseIndex == nil { // only happens if indexing is enabled.
+		n.metrics.wideQuery.ReportError(n.nowFn().Sub(callStart))
+		err := errNamespaceIndexingDisabled
+		sp.LogFields(opentracinglog.Error(err))
+		return err
+	}
+
+	if n.reverseIndex.BootstrapsDone() < 1 {
```

Comment:
This is mirroring how `Query` and `AggregateQuery` do this check; in future may be worth refactoring `NamespaceIndex` to do this check internally, but it won't really save us a lot (we'd be checking that error instead of checking the bootstrap response)

Reply:
Yeah, probably best be to move internally to reverse index.